homeassistant/components/home_connect/binary_sensor.py

```diff
@@ -125,59 +126,41 @@ def available(self) -> bool:
+            status := self.device.appliance.status.get(self.bsh_key, {}).get(ATTR_VALUE)
+        ):
             self._attr_is_on = None
             return
-
-        value = state.get(ATTR_VALUE)
-        if self._false_value_list and self._true_value_list:
-            if value in self._false_value_list:
-                self._attr_is_on = False
-            elif value in self._true_value_list:
-                self._attr_is_on = True
-            else:
-                _LOGGER.warning(
-                    "Unexpected value for HomeConnect %s state: %s", self._type, state
-                )
-                self._attr_is_on = None
-        elif isinstance(value, bool):
-            self._attr_is_on = value
-        else:
-            _LOGGER.warning(
-                "Unexpected value for HomeConnect %s state: %s", self._type, state
-            )
+        if self.entity_description.boolean_map:
+            self._attr_is_on = self.entity_description.boolean_map.get(status)
+        elif status not in [True, False]:
             self._attr_is_on = None
+        else:
+            self._attr_is_on = status
         _LOGGER.debug("Updated, new state: %s", self._attr_is_on)
 
 
-class HomeConnectFridgeDoorBinarySensor(HomeConnectEntity, BinarySensorEntity):
-    """Binary sensor for Home Connect Fridge Doors."""
+class HomeConnectDoorBinarySensor(HomeConnectBinarySensor):
+    """Binary sensor for Home Connect Generic Door."""
 
-    entity_description: HomeConnectBinarySensorEntityDescription
+    _attr_has_entity_name = False
 
     def __init__(
         self,
         device: HomeConnectDevice,
-        entity_description: HomeConnectBinarySensorEntityDescription,
     ) -> None:
         """Initialize the entity."""
-        self.entity_description = entity_description
-        super().__init__(device, entity_description.key, entity_description.desc)
-
-    async def async_update(self) -> None:
-        """Update the binary sensor's status."""
-        _LOGGER.debug(
-            "Updating: %s, cur state: %s",
-            self._attr_unique_id,
-            self.state,
-        )
-        self._attr_is_on = self.entity_description.boolean_map.get(
-            self.device.appliance.status.get(self.bsh_key, {}).get(ATTR_VALUE)
-        )
-        self._attr_available = self._attr_is_on is not None
-        _LOGGER.debug(
-            "Updated: %s, new state: %s",
-            self._attr_unique_id,
-            self.state,
+        super().__init__(
+            device,
+            HomeConnectBinarySensorEntityDescription(
+                key=BSH_DOOR_STATE,
+                device_class=BinarySensorDeviceClass.DOOR,
+                boolean_map={
+                    BSH_DOOR_STATE_CLOSED: False,
+                    BSH_DOOR_STATE_LOCKED: False,
+                    BSH_DOOR_STATE_OPEN: True,
+                },
+            ),
         )
+        self._attr_unique_id = f"{device.appliance.haId}-Door"
```

Comment:
Why don't we use the BSH key here

Reply:
Because a unique id conflict might be raised after #126158 gets merged